setup.py

```diff
@@ -31,1024 +32,1034 @@
-    def_macros += [('HAVE_LZ4', 1), ('HAVE_SNAPPY', 1), ('HAVE_ZLIB', 1),
-                   ('HAVE_ZSTD', 1)]
-
-    # Add extra flags for optimizing shuffle in include Blosc
-    def compiler_has_flags(compiler, flags):
-        with tempfile.NamedTemporaryFile(mode='w', suffix='.c',
-                                         delete=False) as fd:
-            fd.write('int main() {return 0;}')
+    ADDLIBS = [hdf5_package.library_name]
 
-        try:
-            compiler.compile([fd.name], extra_preargs=flags)
-        except Exception:
-            return False
-        else:
-            return True
-        finally:
-            os.remove(fd.name)
-
-    # SSE2
-    if 'sse2' in cpu_flags:
-        print('SSE2 detected')
-        CFLAGS.append('-DSHUFFLE_SSE2_ENABLED')
-        if os.name == 'nt':
-            # Windows always should have support for SSE2
-            # (present in all x86/amd64 architectures since 2003)
-            def_macros += [('__SSE2__', 1)]
-        else:
-            # On UNIX, both gcc and clang understand -msse2
-            CFLAGS.append('-msse2')
+    # List of Blosc file dependencies
+    blosc_sources = ["hdf5-blosc/src/blosc_filter.c"]
+    if 'BLOSC' not in optional_libs:
+        if not os.environ.get('PYTABLES_NO_EMBEDDED_LIBS', None) is None:
+            exit_with_error(
+                "Unable to find the blosc library. "
+                "The embedded copy of the blosc sources can't be used because "
+                "the PYTABLES_NO_EMBEDDED_LIBS environment variable has been "
+                "specified).")
+
+        # Compiling everything from sources
+        # Blosc + BloscLZ sources
         blosc_sources += [f for f in glob.glob('c-blosc/blosc/*.c')
-                          if 'sse2' in f]
-    # AVX2
-    # Detection code for AVX2 only works for gcc/clang, not for MSVC yet
-    if ('avx2' in cpu_flags and
-            compiler_has_flags(compiler, ["-mavx2"])):
-                print('AVX2 detected')
+                        if 'avx2' not in f and 'sse2' not in f]
+        # LZ4 sources
+        blosc_sources += glob.glob('c-blosc/internal-complibs/lz4*/*.c')
+        # Snappy sources
+        blosc_sources += glob.glob('c-blosc/internal-complibs/snappy*/*.cc')
+        # Zlib sources
+        blosc_sources += glob.glob('c-blosc/internal-complibs/zlib*/*.c')
+        # Zstd sources
+        blosc_sources += glob.glob('c-blosc/internal-complibs/zstd*/*/*.c')
+        # Finally, add all the include dirs...
+        inc_dirs += [os.path.join('c-blosc', 'blosc')]
+        inc_dirs += glob.glob('c-blosc/internal-complibs/*')
+        inc_dirs += glob.glob('c-blosc/internal-complibs/zstd*/common')
+        inc_dirs += glob.glob('c-blosc/internal-complibs/zstd*')
+        # ...and the macros for all the compressors supported
+        def_macros += [('HAVE_LZ4', 1), ('HAVE_SNAPPY', 1), ('HAVE_ZLIB', 1),
+                    ('HAVE_ZSTD', 1)]
+
+        #
```

Comment:
@robbmcleod After inspecting this more carefully, shouldn't we add the avx2 source files to the windows ('nt') section above?

Reply:
This isn't fresh in my mind anymore, but isn't the code fragment from lines 885 to 896 doing  that?

```suggestion
             if 'avx2' in cpu_flags:
            print('AVX2 detected')
            if os.name == 'nt':
                if LooseVersion(platform.python_version()) >= LooseVersion('3.5.0'):
                    # Neither MSVC2008 for Python 2.7 or MSVC2010 for Python 3.4 have 
                    # sufficient AVX2 support
                    def_macros += [('__AVX2__', 1)]
            elif compiler_has_flags(compiler, ["-mavx2"]):
                CFLAGS.append('-DSHUFFLE_AVX2_ENABLED')
                CFLAGS.append('-mavx2')
                blosc_sources += [f for f in glob.glob('c-blosc/blosc/*.c')
                                if 'avx2' in f]
```

Reply:
Yeah, but not for `os.name == 'nt'`, so unless I am missing something, the AVX2 support is not reaching to windows.  My current solution (already committed) is to deindent the iclusion of 'avx2' files one level:

```
        if 'avx2' in cpu_flags:
            print('AVX2 detected')
            if os.name == 'nt':
                if LooseVersion(platform.python_version()) >= LooseVersion('3.5.0'):
                    # Neither MSVC2008 for Python 2.7 or MSVC2010 for Python 3.4 have 
                    # sufficient AVX2 support
                    def_macros += [('__AVX2__', 1)]
            elif compiler_has_flags(compiler, ["-mavx2"]):
                CFLAGS.append('-DSHUFFLE_AVX2_ENABLED')
                CFLAGS.append('-mavx2')
            blosc_sources += [f for f in glob.glob('c-blosc/blosc/*.c')  if 'avx2' in f]
```

Makes sense?

Reply:
Oops, correction, the avx2 sources should be included only if the conditions are fulfilled, so better like this:

```
        if 'avx2' in cpu_flags:
            print('AVX2 detected')
            if os.name == 'nt':
                if LooseVersion(platform.python_version()) >= LooseVersion('3.5.0'):
                    # Neither MSVC2008 for Python 2.7 or MSVC2010 for Python 3.4 have 
                    # sufficient AVX2 support
                    def_macros += [('__AVX2__', 1)]
                    blosc_sources += [f for f in glob.glob('c-blosc/blosc/*.c')  if 'avx2' in f]
            elif compiler_has_flags(compiler, ["-mavx2"]):
                CFLAGS.append('-DSHUFFLE_AVX2_ENABLED')
                CFLAGS.append('-mavx2')
                blosc_sources += [f for f in glob.glob('c-blosc/blosc/*.c')  if 'avx2' in f]
```

Agreed?

Reply:
Agreed.